packages/env/src/lib/cast.ts

```diff
@@ -5,9 +5,13 @@ import { guessType } from '../utils/guess-type.js';
 import { getCastFlag } from '../utils/has-cast-prefix.js';
 import { tryJson } from '../utils/try-json.js';
 
-export const cast = (key: string, value: unknown) => {
+export const cast = (key: string, value: unknown): unknown => {
 	const type = getCastFlag(value) ?? getTypeFromMap(key) ?? guessType(value);
 
+	if (getCastFlag(value)) {
+		value = (value as string).split(':').slice(1).join(':');
+	}
```

Comment:
when the prefix contains a  type, remove the first prefix only so that arrays can have inner prefix for each items.
i.e : `array:regex:example\.com$,example2.com` becomes `regex:example\.com$,example2.com`, so that in the array item level cast each element can be casted as well.